### PR TITLE
[NFC][ConstraintSystem] Move some inlined header methods to implementation file 

### PR DESCRIPTION
**Pull request by LucianoPAlmeida**
The initial intention was just to audit and see if some methods could be moved to implementation instead of inline in the header. As commented in another PR normally is recommended to inline only small methods. 
There is not a precise definition of what is a small method, but I thought in propose moving those in this PR. 
While moving I note, that these specific methods have almost exactly the same implementation, so I try to abstract the common logic which seems like an improvement, but TBH not sure if it is worthed nor got much better ... but sending anyway to see what you think cc @xedin @DougGregor

### CODE REVIEW

File: lib/Sema/ConstraintSystem.cpp
Comment (xedin) [inline lib/Sema/ConstraintSystem.cpp:937]: Nit: there is `getDeclOrNull()` method on `OverloadChoice` so checking for `isDecl()` is unnecessary.

File: lib/Sema/ConstraintSystem.cpp
Comment (xedin) [inline lib/Sema/ConstraintSystem.cpp:974]: Let's inline this method into `getPropertyWrapperInformation`?

File: lib/Sema/ConstraintSystem.cpp
Comment (xedin) [inline lib/Sema/ConstraintSystem.cpp:954]: Let's line this into `getStorageWrapperInformation`?

File: lib/Sema/ConstraintSystem.cpp
Comment (xedin) [inline lib/Sema/ConstraintSystem.cpp:991]: Let's inline this into `getWrappedPropertyInformation`?

**Review by xedin — APPROVED** [review]
I like this, I left a couple of minor comments inline.

File: lib/Sema/ConstraintSystem.cpp
Comment (xwu) [inline lib/Sema/ConstraintSystem.cpp:932]:
```suggestion
getPropertyWrapperInformationFromOverload(
```

File: lib/Sema/ConstraintSystem.cpp
Comment (LucianoPAlmeida) [reply lib/Sema/ConstraintSystem.cpp]: Fixed :)

File: lib/Sema/ConstraintSystem.cpp
Comment (LucianoPAlmeida) [reply lib/Sema/ConstraintSystem.cpp:973]: Because of the `None` inline requires an explicit return on the `closure` ... it is ok?

Comment (xedin) [reply lib/Sema/ConstraintSystem.cpp:973]: That’s totally fine.

Comment (LucianoPAlmeida) [reply lib/Sema/ConstraintSystem.cpp]: Right, just fixed :)

**Comment by LucianoPAlmeida** [comment]
@DougGregor as you were one of the last people to modify those methods, pinging for a review :)

**Comment by LucianoPAlmeida** [comment]
@xedin We haven't heard from @DougGregor yet, but since this is a quite simple NFC change I wonder if we could move on? No problem waiting, I'm just doing some checking in some open PR's :)

**Comment by xedin** [comment]
I think this is good to go. Do you want me to merge it (I thought that you going to merge it once you get committer access).

**Comment by LucianoPAlmeida** [comment]
> I think this is good to go. Do you want me to merge it? (I thought that you going to merge it once you get committer access).

Ahh, yes I have the commit access so I can merge it, but still waiting for the access to trigger the CI... so can you kick it then? :)

**Comment by xedin** [comment]
@swift-ci please smoke test

**Comment by LucianoPAlmeida** [comment]
Thanks :) 
Windows build failure seems unrelated, should we kick it again?

**Comment by xedin** [comment]
@swift-ci please smoke test Windows platform

**Comment by xedin** [comment]
If it fails with unrelated problem again - don’t worry and merge.

**Comment by LucianoPAlmeida** [comment]
> If it fails with unrelated problem again - don’t worry and merge.

Oh right, thanks :)

**Pull request merged**
